doc/develop/test/ztest.rst

```diff
@@ -293,18 +293,18 @@ For example:
 Listing Tests
 =============
 
-Tests (test projects) in the Zephyr tree consist of many testcases that run as
+Tests (test applications) in the Zephyr tree consist of many test scenarios that run as
```

Comment:
```suggestion
Tests (test projects) in the Zephyr tree consist of many test scenarios that run as
```
understanding an `application` as a `test project` built under some conditions satisfied for one of its `scenarios`; for Twister, `application` == `instance`.

Reply:
in twister, instance is a scenario, a test application can result in multiple instances. But IMO we should not go into the internals of twister and how it defines things, lets stick to the general, developer facing terminology.

Reply:
agree, an `instance` is Twister internal class corresponding to some `scenario` built.  My point is the difference btw. `test application` and `test project`, where the former might be to name the resulting binary test image.
 
btw, as 'developer facing' twister has its log output with `test scenarios`, `test instances`, `test configurations` and `test cases`, for example:

```
INFO    - 2926 test scenarios (2721 test instances) selected, 2501 configurations skipped (2471 by static filter, 30 at runtime).
INFO    - 220 of 2721 test configurations passed (100.00%), 0 failed, 0 errored, 2501 skipped with 0 warnings in 4468.84 seconds
INFO    - In total 2163 test cases were executed, 12878 skipped on 1 out of total 2 platforms (50.00%)
INFO    - 220 test configurations executed on platforms, 0 test configurations were only built.
```